packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.scss

```diff
@@ -78,20 +78,20 @@
   color: $ms-color-themePrimary;
 }
 
-.root:hover {
-  border-color: $ms-color-themeDarker;
+.root:hover:not(.rootIsDisabled):not(.rootIsActive) {
```

Comment:
My feeling is that once we move to glamor, this is the logic we'll be using to apply this style.